File: C/math/matrix_addition.c
// C program to implement Matrix Addition
#include <stdio.h>
void matrix_add(int [][10],int [][10],int [][10],int,int,int,int);
int main()
{
    int r1, c1, r2, c2;
    printf("Enter the number of rows and columns of the first matrix:  ");
    scanf("%d %d", &r1, &c1);
    printf("Enter the number of rows and columns of the second matrix: ");
    scanf("%d %d", &r2, &c2);

    //If the given matrices differ in thier number of rows and columns, they cannot be added
    if ((r1 != r2) || (c1 != c2))
    {
        printf("Given Matrices cannot be added!!!");
        return 0;
    }
    int A[10][10], B[10][10], C[10][10];
    // Input the values of the matrices
    printf("Enter the values of the first matrix\n");
    for (int i = 0; i < r1; i++)
    {
        for (int j = 0; j < c1; j++)
            scanf("%d", &A[i][j]);
    }
    printf("Enter the values of the second matrix\n");
    for (int i = 0; i < r2; i++)
    {
        for (int j = 0; j < c2; j++)
            scanf("%d", &B[i][j]);
    }
    matrix_add(C,A,B,r1,r2,c1,c2);
    printf("The resultant matrix is:\n");
    for (int i = 0; i < r2; i++)
    {
        for (int j = 0; j < c2; j++)
            printf("%d ", C[i][j]);
        printf("\n");
    }
    return 0;
}
void matrix_add(int C[][10],int A[][10],int B[][10],int r1,int r2,int c1,int c2)
{
    for (int i = 0; i < r1; i++)
    {
        for (int j = 0; j < c2; j++)
        {
            // Add the corresponding values of both the matrices
            C[i][j] = A[i][j] + B[i][j];
        }
    }
}

/*
Time Complexity: O(r * c), where 'r' is the number of rows and 'c' is the number of columns 
Space Complexity: O(r * c)

SAMPLE INPUT AND OUTPUT

Enter the number of rows and columns of the first matrix:  2 2
Enter the number of rows and columns of the second matrix:  2 2
Enter the values of the first matrix
2 2
2 2
Enter the values of the second matrix
2 2
2 2
The resultant matrix is:
4 4
4 4

*/

File: Python/math/Matrix_addition.py
"""
In mathematics, matrix addition is the operation of adding two matrices by adding 
the corresponding entries together.This program takes two matrices of order r*c and 
stores it in two-dimensional array. Then, the program adds these two matrices and 
displays it on the screen.
"""

Row_1 = int(input("Enter the number of rows for first matrix : "))
Col_1 = int(input("Enter the number of columns for first matrix : "))

matrix_1 = []
print("Enter the entries rowwise:")
for i in range(Row_1):
    matrix_1.append([int(x) for x in input().split(" ")])

Row_2 = int(input("Enter the number of rows for second matrix : "))
Col_2 = int(input("Enter the number of columns for second matrix : "))

matrix_2 = []
print("Enter the entries rowwise:")
for i in range(Row_2):
    matrix_2.append([int(x) for x in input().split(" ")])

result = [[0 for i in range(Col_2)] for j in range(Row_1)]
# Adding both matrices and storing in result
for i in range(Row_1):
    for j in range(Col_2):
        result[i][j] = matrix_1[i][j] + matrix_2[i][j]

print("The result of the matrix Addition is")
for i in range(Row_1):
    for j in range(Col_2):
        print(result[i][j], end=" ")
    print()

"""
Sample I/O :

Enter the number of rows for first matrix : 3
Enter the number of columns for first matrix : 3
Enter the entries rowwise:
1 2 3
4 5 6
7 8 9
Enter the number of rows for second matrix : 3
Enter the number of columns for second matrix : 3
Enter the entries rowwise:
1 2 3
4 5 6
7 8 9
The result of the matrix Addition is
2 4 6
8 10 12
14 16 18

Time complexity : O(n^2)
Space complexity : O(n^2)
"""
